src/main/java/de/zalando/aruha/nakadi/repository/kafka/KafkaLocationManager.java

```diff
@@ -0,0 +1,126 @@
+package de.zalando.aruha.nakadi.repository.kafka;
+
+import java.io.UnsupportedEncodingException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Properties;
+
+import javax.annotation.PostConstruct;
+
+import org.apache.curator.framework.CuratorFramework;
+import org.json.JSONException;
+import org.json.JSONObject;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.scheduling.annotation.Scheduled;
+
+import de.zalando.aruha.nakadi.repository.zookeeper.ZooKeeperHolder;
+
+public class KafkaLocationManager {
+
+    private static final Logger LOG = LoggerFactory.getLogger(KafkaLocationManager.class);
+
+    private final String _BROKERS_IDS_PATH = "/brokers/ids";
+
+    @Autowired
+    private ZooKeeperHolder zkFactory;
+
+    private List<Broker> kafkaBrokerList;
+
+    private Properties kafkaProperties;
+
+    class Broker implements Comparable<Broker> {
+        final String host;
+        final Integer port;
+
+        public Broker(final String host, final Integer port) {
+            this.host = host;
+            this.port = port;
+        }
+
+        @Override
+        public int compareTo(Broker other) {
+            final int result = host.compareTo(other.host);
+            return result == 0 ? port.compareTo(other.port) : result;
+        }
+
+        @Override
+        public boolean equals(Object other) {
+            if (other == this)
+                return true;
+            if (!(other instanceof Broker))
+                return false;
+            return this.compareTo((Broker)other) == 0;
+        }
+
+        public String toString() {
+            return this.host + ":" + this.port;
+        }
+    }
+
+    private Broker getBroker(final byte[] data) throws JSONException, UnsupportedEncodingException {
+        final JSONObject json = new JSONObject(new String(data, "UTF-8"));
+        final String host = json.getString("host");
+        final Integer port = json.getInt("port");
+        return new Broker(host, port);
+    }
+
+    private List<Broker> fetchBrokers() {
+        final List<Broker> brokers = new ArrayList<Broker>();
+        try {
+            final CuratorFramework curator = zkFactory.get();
+            for (final String brokerId : curator.getChildren().forPath(_BROKERS_IDS_PATH)) {
+                try {
+                    final byte[] brokerData = curator.getData().forPath(_BROKERS_IDS_PATH + "/" + brokerId);
+                    brokers.add(getBroker(brokerData));
+                } catch (Exception e) {
+                    LOG.info("Failed to fetch connection string for broker {}: {}", brokerId, e);
```

Comment:
I think it is not working that way. The interface of slf4j is:

```
info(String msg); 
info(String format, Object arg); 
info(String format, Object... args);           
info(String msg, Throwable t);
```

So if you pass in two parameters it will treat the Throwable as an object and thus calls e.toString() and loose the stacktrace.